modules/utxo-lib/src/bitgo/signature.ts

```diff
@@ -102,6 +103,21 @@ export function getSignatureVerifications<TNumber extends number | bigint>(
     }
   }
 
+  if (
+    parsedScript.scriptType !== 'taprootKeyPathSpend' &&
+    parsedScript.scriptType !== 'taprootScriptPathSpend' &&
+    prevOutputs
+  ) {
+    const prevoutScript = prevOutputs[inputIndex].script;
+
+    const output = getOutputScript(parsedScript.scriptType, parsedScript.pubScript);
+    if (!prevoutScript.equals(output)) {
+      throw new Error(
+        `prevout script ${prevoutScript.toString('hex')} does not match computed script ${output.toString('hex')}`
+      );
```

Comment:
We're gonna be computing the script every time we verify the existing sigs?
Although most likely a non-issue, but off the top of your head, how much overhead is this gonna add in the regular signing flow?

Reply:
I think it is very small: this sort of check happens in the psbt flow by default without significant impact